### ISSUE TITLE
API rewrite

### ISSUE THREAD

**Issue by mitchelloharawild** [mitchelloharawild/icons#19]
cc: @earowang, @emitanaka

Rewrite of the interface to introduce support for user defined icon libraries, and extensibility support (hopefully improving [anicon](https://github.com/emitanaka/anicon)). Along with the rewrite, the package will convert from icon fonts to SVG icons.

I imagine that anicon may be better suited as a general purpose animation library for rmarkdown documents. This is something I'd like to work on later this month.

---

## Icon styling
`modify(icon, colour = "red", rotate = 90)`

## Icon stacking
`icon1 + icon2`

## Icon animation
`animate(icon, animation)`

## New icon library
```r
my_icon <- new_icon(svgfolder)
my_icon$icon_name(...)
my_icon(icon_name)
```

## Example implementation of functions with `$` methods
```r
my_icon <- structure(function(...){
  return(..1)
}, class = "icon")

`$.icon` <- function(lib, icon){
  return(deparse(icon))
}

my_icon$rocket
my_icon("rocket")
```

**Comment by emitanaka** [comment]
Yes, I like this! Will be piping compatible! Was thinking of something similar so good to see we are on the same page :) Icon stacking needs a bit more thinking. I'm fine with the API but under the hoods will be a pain.

**Comment by mitchelloharawild** [comment]
Piping is the idea :smile: 
It is also nice because the animation and styles are not necessarily specific to icons. We could also have the modify and animation functionality work on text (perhaps better suited to a separate package).

I'm happy to think more about icon stacking, it may be better to have it in a more pipe-able UI. For example:
```r
fa$moon %>%
  stack(
    fa$rocket %>%
      modify(size = 0.5)
  )

**Comment by emitanaka** [comment]
That's one idea. I do like the `icon1 + icon2` style. Reminds of [patchwork](https://github.com/thomasp85/patchwork). I'll think more about it too before I see you in person!

PS: nice user2018 talk!

**Comment by mitchelloharawild** [comment]
Resolved by #31
